pilot/pkg/config/kube/clustertrustbundle/controller.go

```diff
@@ -0,0 +1,142 @@
+
+	// maxRetries is the number of times a ClusterTrustBundle reconciliation will be retried before it is dropped
+	// out of the queue.
+	// With the current rate-limiter in use (5ms*2^(maxRetries-1)) the following numbers represent the
+	// sequence of delays between successive queuing of a namespace.
+	//
+	// 5ms, 10ms, 20ms, 40ms, 80ms
+	maxRetries = 5
+)
+
+// ClusterTrustBundleController manages the lifecycle of the Istio root certificate in ClusterTrustBundle
+// It watches Istio's KeyCert bundle as well as ClusterTrustBundles in the cluster, and it updates ClusterTrustBundles whenever
+// the certificate is rotated. Later on, we might want to add reading from other ClusterTrustBundles as well
+type ClusterTrustBundleController struct {
+	client              kube.Client
+	caBundleWatcher     *keycertbundle.Watcher
+	queue               controllers.Queue
+	clustertrustbundles kclient.Client[*certificatesv1alpha1.ClusterTrustBundle]
+	stopChan            chan struct{}
+}
+
+// NewClusterTrustBundleController creates a new ClusterTrustBundleController
+func NewClusterTrustBundleController(kubeClient kube.Client, caBundleWatcher *keycertbundle.Watcher) *ClusterTrustBundleController {
+	c := &ClusterTrustBundleController{
+		client:          kubeClient,
+		stopChan:        make(chan struct{}),
+		caBundleWatcher: caBundleWatcher,
+	}
+
+	c.queue = controllers.NewQueue("clustertrustbundle controller",
+		controllers.WithReconciler(c.reconcileClusterTrustBundle),
+		controllers.WithMaxAttempts(maxRetries))
+
+	c.clustertrustbundles = kclient.NewFiltered[*certificatesv1alpha1.ClusterTrustBundle](kubeClient, kclient.Filter{
+		FieldSelector: "metadata.name=" + istioClusterTrustBundleName,
+		ObjectFilter:  kubeClient.ObjectFilter(),
+	})
+
+	c.clustertrustbundles.AddEventHandler(controllers.FilteredObjectSpecHandler(c.queue.AddObject, func(o controllers.Object) bool { return true }))
+
+	return c
+}
+
+// startCaBundleWatcher listens for updates to the CA bundle and queues a reconciliation of the ClusterTrustBundle
+func (c *ClusterTrustBundleController) startCaBundleWatcher(stop <-chan struct{}) {
+	id, watchCh := c.caBundleWatcher.AddWatcher()
+	defer c.caBundleWatcher.RemoveWatcher(id)
+	for {
+		select {
+		case <-watchCh:
+			c.queue.AddObject(&certificatesv1alpha1.ClusterTrustBundle{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: istioClusterTrustBundleName,
+				},
+			})
+		case <-stop:
+			return
+		}
+	}
+}
+
+func (c *ClusterTrustBundleController) reconcileClusterTrustBundle(o types.NamespacedName) error {
+	if o.Name == istioClusterTrustBundleName {
+		return c.updateClusterTrustBundle(c.caBundleWatcher.GetCABundle())
+	}
+	return nil
+}
+
+// Run starts the controller
+func (c *ClusterTrustBundleController) Run(stopCh <-chan struct{}) {
+	if !kube.WaitForCacheSync("clustertrustbundle controller", stopCh, c.clustertrustbundles.HasSynced) {
+		return
+	}
+	go c.startCaBundleWatcher(stopCh)
+	defer close(c.stopChan)
```

Comment:
nit: c.stopChan is not used anywhere except here to close

Reply:
oops, leftover from previous iteration. fixed